src/lib/middleware/validators.js

```diff
@@ -0,0 +1,39 @@
+export const categoryValidator = store => next => action => {
+  let actionType = action.type.split('_');
+  if (actionType[0] === 'CATEGORY'){
+    try{
+      console.log(action);
+
+      let result = next(action);
+
+      if (!action.payload.title) {throw new Error('no title provided')}
+      else if (!action.payload.budget) {throw new Error('no budget provided')}
+      else if (!action.payload.id) {throw new Error('no id provided')}
+
+      return result;
+    } catch (err) {
+      err.action = action
+      throw err;
+    }
+  }
+  return next(action);
+}
+
+export const expenseValidator = store => next => action => {
+  let actionType = action.type.split('_');

```

Comment:
This works as long as your action names stay consistent with an underscore, but if you ever add an action type without an underscore (say "LOGIN" or "LOGOUT") then you will likely have errors thrown on the next line when you attempt to access actionType[0].

Instead, consider using some string methods to find a substring in the action type. For instance, String.prototype.includes or String.prototype.startsWith.